classes/ActionScheduler_OptionLock.php

```diff
@@ -24,7 +24,25 @@ class ActionScheduler_OptionLock extends ActionScheduler_Lock {
 	 * @bool True if lock value has changed, false if not or if set failed.
 	 */
 	public function set( $lock_type ) {
-		return update_option( $this->get_key( $lock_type ), time() + $this->get_duration( $lock_type ) );
+		global $wpdb;
+
+		$existing_lock_value  = $this->get_or_create_lock( $lock_type );
+		$expiration_timestamp = $this->get_expiration_from( $existing_lock_value );
```

Comment:
Is there a reason we could not just get expiration and, depending on the result, INSERT or UPDATE?

It seems unnecessary to potentially INSERT a lock option value in order to conditionally UPDATE against that value later on. Currently, this requires three queries:
1. INSERT lock if none exists
2. SELECT lock value
3. UPDATE if old value is expected value

If we conditionally INSERT or UPDATE, it would require two queries:
1. SELECT lock value if any
2. INSERT or UPDATE depending on whether there is an existing and expired lock value

Reply:
Definitely ... a little unsure why I did things this way. In any case, mind taking a further peek (especially at https://github.com/woocommerce/action-scheduler/pull/967/commits/895e8d0076590ec1fe55fb1fe52af65e3d4a8340)?

Reply:
> Definitely ... a little unsure why I did things this way. 

I've had similar experiences :)

> In any case, mind taking a further peek (especially at https://github.com/woocommerce/action-scheduler/commit/895e8d0076590ec1fe55fb1fe52af65e3d4a8340)?

Sure thing!